src/TLC.sol

```diff
@@ -0,0 +1,308 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+
+import {Ownable} from "openzeppelin-contracts/access/Ownable.sol";
+import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "openzeppelin-contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {Operators} from "./common/access/Operators.sol";
+
+contract TLC is Ownable, Operators {
+
+    using SafeERC20 for IERC20;
+
+    struct Position {
+        uint256 collateralAmount;
+        uint256 debtAmount;
+        uint256 createdAt;
+    }
+
+    // Collateral Parameters
+
+    /// Supported collateral token address
+    address public collateralAddress;
+
+    /// Collateral token price 
+    uint256 public collateralPrice;
+
+    /// Requited collateral backing to not be in bad debt
+    uint256 public collateralizationRatio;
+
+    /// Total collateral posted
+    uint256 public collateralBalance;
+
+    //// Total debt taken out
+    uint256 public debtBalance;
+
+    /// Fixed interest rate
+    uint256 public interestRate;
+
+    /// Amount in seconds for interest to accumulate
+    uint256 public interestRatePeriod;
+
+    /// Fee for taking out a loan
+    uint256 public originationFee;
+
+    /// Fee charged for debtor liquidation
+    uint256 public liquidationFee;
+
+    /// Address to send bad debt collateral
+    address public debtCollector;
+
+    /// Debt parameters
+
+    /// Debt token address
+    address public debtAddress;
+
+    /// Debt token price
+    uint256 public debtPrice;
```

Comment:
Treasury Price Index changes over time - ie goes up over time. Won't we need a setter on this to update too?